### PR TITLE
Feature/1698/tests

### PR DESCRIPTION
**Pull request by garyluu**
Tests for #1698

### CODE REVIEW

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java [hunk 1]
```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
...
+     * @throws Exception
+     */
+    private void registerNewUsersWithExisting(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGitHub(CUSTOM_USERNAME1);
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        // Cannot create new user with the same Google account
+        try {
+            unAuthenticatedTokensApi.addGoogleToken(satellizerJSONForRegistration);
+            Assert.fail();
+        } catch (Exception e){
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: Use the narrowest exception required, I think in this case you just need ApiException

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java [hunk 2]
```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
...
+            Assert.assertEquals("User already exists, cannot register new user", e.getMessage());;
+            // Call should fail
+        }
+
+        // Cannot create new user with the same GitHub account
+        try {
+            unAuthenticatedTokensApi.addToken(satellizerJSONForRegistration);
+            Assert.fail();
+        } catch (Exception e){
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: same

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java [hunk 3]
```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
...
+     */
+    private void addUnavailableGoogleTokenToGitHubUser() throws Exception {
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        TokensApi otherUserTokensApi = new TokensApi(getWebClient(true, GITHUB_ACCOUNT_USERNAME));
+        // Cannot add token to other user with the same Google account
+        try {
+            otherUserTokensApi.addGoogleToken(satellizerJSON);
+            Assert.fail();
+        } catch (Exception e){
+            Assert.assertTrue(e.getMessage().contains("already exists."));;
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: Leave out the punctuation at minimum
Test based on HTTP code would be better. The message might easily change if for example we change it to something like "already exists, you have probably registered"

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java [hunk 4]
```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
...
+     * @throws Exception
+     */
+    private void addUnavailableGitHubTokenToGoogleUser(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGitHub(CUSTOM_USERNAME1);
+        TokensApi otherUserTokensApi = new TokensApi(getWebClient(true, GOOGLE_ACCOUNT_USERNAME2));
+        try {
+            otherUserTokensApi.addGithubToken("potato");
+            Assert.fail();
+        } catch (Exception e){
+            Assert.assertTrue(e.getMessage().contains("already exists."));;
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: same

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java (hunk 1 above)
Comment (garyluu) [reply dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: It's the mockGitHub and powermock that's throwing the Exception.

Comment (denis-yuen) [reply dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: @garyluu If that's true, the test is probably not quite testing what we want. 
We want to be testing against exceptions that are thrown in our code (or the swagger generated classes when expected). We shouldn't be testing in what situations powermock crashes with an exception. Feel free to swing by if you need clarification

Comment (garyluu) [reply dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java]: nvm, was looking at wrong line

**Comment by denis-yuen** [comment]
Thanks https://www.javaworld.com/article/2077500/testing-debugging/java-tip-134--when-catching-exceptions--don-t-cast-your-net-too-wide.html if you're curious

**Pull request merged**